File: jddata/app/src/main/java/com/example/jddata/shelldroid/Env.java
package com.example.jddata.shelldroid;

public class Env implements Cloneable {
    public String id;
    public String envName;
    public String appName;
    public String pkgName;
    public boolean active;
    public String deviceId;
    public String phoneNumber;
    public String networkCountryIso;
    public String networkOperator;
    public String simSerialNumber;
    public String buildBoard;
    public String buildModel;
    public String buildManufacturer;
    public String buildId;
    public String buildDevice;
    public String buildSerial;
    public String buildBrand;
    public String androidId;
    public Location location;

    @Override
    public String toString() {
        return "Env{" +
                "id='" + id + '\'' +
                ", envName='" + envName + '\'' +
                ", appName='" + appName + '\'' +
                ", pkgName='" + pkgName + '\'' +
                ", active=" + active +
                ", deviceId='" + deviceId + '\'' +
                ", phoneNumber='" + phoneNumber + '\'' +
                ", networkCountryIso='" + networkCountryIso + '\'' +
                ", networkOperator='" + networkOperator + '\'' +
                ", simSerialNumber='" + simSerialNumber + '\'' +
                ", buildBoard='" + buildBoard + '\'' +
                ", buildModel='" + buildModel + '\'' +
                ", buildManufacturer='" + buildManufacturer + '\'' +
                ", buildId='" + buildId + '\'' +
                ", buildDevice='" + buildDevice + '\'' +
                ", buildSerial='" + buildSerial + '\'' +
                ", buildBrand='" + buildBrand + '\'' +
                ", androidId='" + androidId + '\'' +
                ", location=" + location +
                '}';
    }

    @Override
    protected Env clone() throws CloneNotSupportedException {
        return (Env) super.clone();
    }
}

File: jddata/app/src/main/java/com/example/jddata/shelldroid/Env.kt
package com.example.jddata.shelldroid

import com.example.jddata.Entity.EnvActions
import java.io.Serializable

class Env : Cloneable, Serializable {
    private val serialVersionUID = 1L

    var id: String? = null
    var envName: String? = null
    var appName: String? = null
    var pkgName: String? = null
    var active: Boolean = false
    var imei: String? = null
    var createTime: String? = null
    var locationNo: String? = null
    var locationName: String? = null
    var longitude: Double? = null
    var latitude: Double? = null
    var observation: String? = null
    var day9: String? = null
    var envActions: EnvActions? = null
    var move: String? = null

    @Throws(CloneNotSupportedException::class)
    public override fun clone(): Env {
        return super.clone() as Env
    }

    override fun toString(): String {
        return "Env(id=$id, envName=$envName, appName=$appName, pkgName=$pkgName, active=$active, imei=$imei, createTime=$createTime, locationNo=$locationNo, locationName=$locationName, longitude=$longitude, latitude=$latitude, observation=$observation)"
    }

}
